### PR TITLE
HttpListenerRequest Uri is now unescaped.

### PR DESCRIPTION
**Pull request by esdrubal**
**Fixes:** [#29927](https://bugzilla.xamarin.com/show_bug.cgi?id=29927)

### CODE REVIEW

File: mcs/class/System/ReferenceSources/HttpSysSettings.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace System.Net
+{
+	class HttpSysSettings
+    {
```
Comment (knocte) [inline mcs/class/System/ReferenceSources/HttpSysSettings.cs]: Wrong indentation here.

Comment (esdrubal) [inline mcs/class/System/ReferenceSources/HttpSysSettings.cs]: Fixed. Thanks.

File: mcs/class/System/ReferenceSources/HttpSysSettings.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+namespace System.Net
+{
+	class HttpSysSettings
```
Comment (marek-safar) [inline mcs/class/System/ReferenceSources/HttpSysSettings.cs]: Should be static

File: mcs/class/System/ReferenceSources/HttpSysSettings.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+	{
+		public static bool EnableNonUtf8 = true;
+		public static bool FavorUtf8 = true;
```
Comment (marek-safar) [inline mcs/class/System/ReferenceSources/HttpSysSettings.cs]: Both should be const

File: mcs/class/System/System.dll.sources [hunk 4]
```diff
@@ -1121,4 +1123,3 @@ ReferenceSources/Win32Exception.cs
 ../../../external/referencesource/System/compmod/system/codedom/compiler/ICodeParser.cs
 ../../../external/referencesource/System/compmod/system/codedom/compiler/IndentTextWriter.cs
 ../../../external/referencesource/System/compmod/system/codedom/compiler/LanguageOptions.cs
-
```
Comment (marek-safar) [inline mcs/class/System/System.dll.sources]: This is not the only .sources file which needs to be updated

File: mcs/class/System/ReferenceSources/HttpSysSettings.cs (hunk 3 above)
Comment (esdrubal) [inline mcs/class/System/ReferenceSources/HttpSysSettings.cs]: Class is now static and fields are now const. Thanks

**Pull request merged**